### PR TITLE
Module naming cleanup

### PR DESCRIPTION
**Pull request by jkwatson**
Rename the components module to just refer to http, now that gson is gone.
Also includes some doc updates.

### CODE REVIEW

File: THIRD_PARTY_NOTICES.md [hunk 1]
```diff
@@ -0,0 +1,180 @@
+# Third Party Notices
+
+The New Relic Java Telemetry SDK uses source code from third party libraries which carry
+their own copyright notices and license terms. These notices are provided
+below.
+
+In the event that a required notice is missing or incorrect, please notify us
+either by [opening an issue](https://github.com/newrelic/c-sdk/issues/new), or
```
Comment (breedx-nr) [inline THIRD_PARTY_NOTICES.md]: whoopsie

File: THIRD_PARTY_NOTICES.md [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+by e-mailing [open-source@newrelic.com](mailto:open-source@newrelic.com).
+
+## Contents
+
+* [gson](#gson)
+
+## gson
+
+This product includes source derived from
+[gson](https://github.com/DaveGamble/cJSON), distributed under the
```
Comment (breedx-nr) [inline THIRD_PARTY_NOTICES.md]: whoopsie

File: telemetry-http-okhttp/README.md [hunk 3]
```diff
@@ -1,4 +1,4 @@
-This module contains reference implementations of the required interfaces for the SDK to function.
+This module contains reference implementations of the required http client for the SDK to function.
```
Comment (breedx-nr) [inline telemetry-http-okhttp/README.md]: nitpick: should be singular `a reference implementation`

Comment (jkwatson) [reply telemetry-http-okhttp/README.md]: :ack:

Comment (jkwatson) [reply telemetry-http-okhttp/README.md]: check the changes

**Pull request merged**